app/questionnaire/navigation.py

```diff
@@ -141,7 +141,7 @@ def _generate_repeated_items(self, link_names, first_location, completed_id, no_
             location.group_instance = i
             is_current_group_instance = is_current_group and i == current_group_instance
 
-            if link_names:
+            if link_names and link_names.get(i):
```

Comment:
Or since this only appears to be an issue when hitting `save and continue later` on the household question at the moment, could just deal with this specific case and delete any answers that have empty values for this question when save and continue later is pressed?